src/reporter.ts

```diff
@@ -153,18 +157,33 @@ export function getPrioritizedSpans(spans: any[], maxSendBytes: number): any[] {
 export const forgeAndScrubRequestBody = (
   spans,
   maxSendBytes,
-  maxSendBytesOnError
+  maxSendBytesOnError,
+  shouldTryZip: boolean = false
 ): string | undefined => {
   const maxRequestSize = spans.some(spanHasErrors) ? maxSendBytesOnError : maxSendBytes;
   const start = new Date().getTime();
   const beforeLength = spans.length;
   const originalSize = spans.length;
   const size = getJSONBase64Size(spans);
+  if (spans.length == 0) {
+    return undefined;
+  }
 
   if (
     (!isPruneTraceOff() && spans.length > NUMBER_OF_SPANS_IN_REPORT_OPTIMIZATION) ||
     size > maxSendBytes
   ) {
+    if (shouldTryZip) {
+      logger.debug(`Trying to zip spans, size [${size}], bigger than: [${maxRequestSize}]`);
+      const zippedSpans = gzipSync(JSON.stringify(spans)).toString('base64');
```

Comment:
`gzipSync` blocks the main thread, might pose a performance hit and choke the lambda - can we use the async version? maybe we can zip the payload during the sending part, not when forging the body?